fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/common/SchedulerJobHelper.java

```diff
@@ -122,45 +139,112 @@ private static String runSchedulerJobAsJSON() {
+        assertEquals("Verifying Last Scheduler Job Status", "success",
+                jobHistoryData.get(jobHistoryData.size() - 1).get("status"));
+    }
 
-                break;
+    /**
+     * Launches a Job and awaits its completion.
+     * @param jobName displayName (see {@link org.apache.fineract.infrastructure.jobs.service.JobName}) of Scheduler Job
+     *
+     * @author Michael Vorburger.ch
+     */
+    public void executeAndAwaitJob(String jobName) {
+        Duration TIMEOUT = Duration.ofSeconds(30);
+        Duration PAUSE = Duration.ofMillis(500);
+        DateTimeFormatter df = DateTimeFormatter.ISO_INSTANT; // FINERACT-926
+        Instant beforeExecuteTime = now().truncatedTo(ChronoUnit.SECONDS);
+
+        // Stop the Scheduler while we manually trigger execution of job, to avoid side effects and simplify debugging when readings logs
+        updateSchedulerStatus(false);
+
+        // Executing Scheduler Job
+        int jobId = getSchedulerJobIdByName(jobName);
+        runSchedulerJob(jobId);
+
+        // Await JobDetailData.lastRunHistory [JobDetailHistoryData] jobRunStartTime >= beforeExecuteTime (or timeout)
+        await().atMost(TIMEOUT).pollInterval(PAUSE).until(jobLastRunHistorySupplier(jobId), lastRunHistory -> {
+            String jobRunStartText = lastRunHistory.get("jobRunStartTime");
+            if (jobRunStartText == null) {
+                return false;
             }
+            Instant jobRunStartTime = df.parse(jobRunStartText, Instant::from);
+            return jobRunStartTime.equals(jobRunStartTime) || jobRunStartTime.isAfter(beforeExecuteTime);
+        });
+
+        // Await JobDetailData.lastRunHistory [JobDetailHistoryData] jobRunEndTime to be both set and >= jobRunStartTime (or timeout)
+        Map<String, String> finalLastRunHistory = await().atMost(TIMEOUT).pollInterval(PAUSE).until(jobLastRunHistorySupplier(jobId), lastRunHistory -> {
+            String jobRunEndText = lastRunHistory.get("jobRunEndTime");
+            if (jobRunEndText == null) {
+                return false;
+            }
+            Instant jobRunEndTime = df.parse(jobRunEndText, Instant::from);
+            Instant jobRunStartTime = df.parse(lastRunHistory.get("jobRunStartTime"), Instant::from);
+            return jobRunEndTime.equals(jobRunStartTime) || jobRunEndTime.isAfter(jobRunStartTime);
+        });
+
+        // Verify triggerType
+        assertThat(finalLastRunHistory.get("triggerType"), is("application"));
+
+        // Verify status & propagate jobRunErrorMessage and/or jobRunErrorLog (if any)
+        String status = finalLastRunHistory.get("status");
+        if (!status.equals("success")) {
+            fail("Job status is not success: " + finalLastRunHistory.toString());
         }
+
+        // PS: Checking getSchedulerJobHistory() [/runhistory] is pointless, because the lastRunHistory JobDetailHistoryData is already part of JobDetailData anyway.
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
Would be great to get rid of the SuppressWarnings... I wonder if there's any clever solution to achieve that?

Reply:
As far as I can see, no, not easily unfortunately. Why? Vaguely related to what I blabbered on about in FINERACT-949 re. JSON mapping. I do want to think further about that some fine day, but not as part of this PR. OK to resolve this point?

Reply:
Of course - makes sense. One solution that I thought could work for FINERACT-949 in general would be to use the responseSpec for checking response correctness (instead of looking at return values). But let's continue the discussion under that issue. So OK to resolve.